Methane_MarkDown.Rmd

```diff
@@ -111,3 +110,206 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
+  result
+}
+```
+
+Then we tell hector to run a range of values. In this case we ask for 20% above and bellow the set value of `NATURAL_CH4`(341Tg) at intervals of 5 +/-20%. These set values are labelled as `ch4-seq`.
+
+```{r sen-param}
+run_with_param_range <- function(core, parameter, values) {
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+valuen<- default %>%
+  select(value)
+meann<- valuen$value
+ch4_low <- meann * (1 - 0.2)
+ch4_high <- meann * (1 + 0.2)
+ch4_seq <- seq(ch4_low, ch4_high, 5)
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), ch4_seq)
+```
+
+Once we have our differing model outputs we can create a graph. Using the graph function `facet_wrap`to separate the results based on variable. 
+
+```{r sen-graph}
+ggplot(sensitivity_emission) +
+  aes(x = year, y = value, color = parameter_value, group = parameter_value) +
+  geom_line() +
+  facet_wrap(~variable, scales = "free_y") +
+  labs(color = "NATURAL CH4 (Tg)") +
+  scale_color_viridis_c() 
+```
+
+### Adding a Histogram 
+
+Using the same `Natural_CH4` data, we can also make a histogram plot. Since we already made `ch4_low` and `ch4_high` with levels +/-20%, we can re-purpose them for this graph.
+
+```{r rnorm1}
+sdn<- meann*0.2
+data<- data.frame(value= rnorm(1000, meann, sdn))
+```
+
+Then we can use `ggplot2` to create a histogram with the randomized values.
+```{r hist}
+ggplot(data, aes(x=value))+
+  geom_histogram(binwidth = 1)+
+  labs( x= "Natural CH4 Level", y= "count", main= "Natural CH4 Histogram")
+invisible(shutdown(core))
+```
+
+
+## Multiple Parameter Run
+
+Working with four parameters `NATURAL_CH4`, `PREINDUSTRIAL_CH4`, `LIFETIME_SOIL` and `LIFETIME_STRAT` to create a sensitivity analysis. First we run all of them through `fetchvars`.
+
+```{r core}
+core <- newcore(ini_file)
+run(core)
+results <- fetchvars(core, NA, c(NATURAL_CH4(), PREINDUSTRIAL_CH4(),LIFETIME_SOIL(), LIFETIME_STRAT()))
+```
+
+We then create two data frames- one for each parameter. First isolating the variable, and then calculating mean and standard deviation for the variable's value. The using `rnorm` we create 10 values pulled from the normal distribution, putting our results in `data.frame`.
+
+```{r rnorm2}
+nat_value<- results %>%
+  filter(variable== "CH4N") 
+nat_mean<- nat_value$value
+nat_sd<- nat_mean*0.2
+nat_data<- data.frame(value= rnorm(10, nat_mean, nat_sd))
+
+pre_value<- results %>%
+  filter(variable== "M0") 
+pre_mean<- pre_value$value
+pre_sd<- pre_mean*0.2
+pre_data<- data.frame(value= rnorm(10, pre_mean, pre_sd))
+
+soil_value<- results %>%
+  filter(variable== "Tsoil") 
+soil_mean<- soil_value$value
+soil_sd<- soil_mean*0.2
+soil_data<- data.frame(value= rnorm(10, soil_mean, soil_sd))
+
+strat_value<- results %>%
+  filter(variable== "Tstrat") 
+strat_mean<- strat_value$value
+strat_sd<- strat_mean*0.2
+strat_data<- data.frame(value= rnorm(10, strat_mean, strat_sd))
```

Comment:
Rather than do `norm(10, ...)` all the way through this section, I would define a variable `N_RUNS` at the top and use it throughout. That way, to change the number of runs you just need to change that setting in one place.

Reply:
this is much better thanks!